cropper/src/main/java/com/canhub/cropper/CropImageActivity.kt

```diff
@@ -323,4 +349,168 @@ open class CropImageActivity :
+     * Create a chooser intent to select the source to get image from.<br></br>
+     * The source can be camera's (ACTION_IMAGE_CAPTURE) or gallery's (ACTION_GET_CONTENT).<br></br>
+     * All possible sources are added to the intent chooser.
+     *
+     * @param context          used to access Android APIs, like content resolve, it is your
+     * activity/fragment/widget.
+     * @param title            the title to use for the chooser UI
+     * @param includeCamera    if to include camera intents
+     * @param includeGallery if to include Gallery app intents
+     * @param includeDocuments if to include KitKat documents activity containing all sources
+     */
+    private fun getPickImageChooserIntent(
+        context: Context,
+        title: CharSequence?,
+        includeCamera: Boolean,
+        includeGallery: Boolean,
+        includeDocuments: Boolean
+    ): Intent {
+        val allIntents: MutableList<Intent> = ArrayList()
+        val packageManager = context.packageManager
+        // collect all camera intents if Camera permission is available
+        if (!isExplicitCameraPermissionRequired(context) && includeCamera) {
+            allIntents.addAll(getCameraIntents(context, packageManager))
+        }
+        if (includeGallery) {
+            var galleryIntents =
+                getGalleryIntents(packageManager, Intent.ACTION_GET_CONTENT, includeDocuments)
+            if (galleryIntents.isEmpty()) {
+                // if no intents found for get-content try pick intent action (Huawei P9).
+                galleryIntents =
+                    getGalleryIntents(packageManager, Intent.ACTION_PICK, includeDocuments)
+            }
+            allIntents.addAll(galleryIntents)
+        }
+        val target: Intent
+        if (allIntents.isEmpty()) {
+            target = Intent()
+        } else {
+            target = Intent(Intent.ACTION_CHOOSER, MediaStore.Images.Media.EXTERNAL_CONTENT_URI)
+            if (includeGallery) {
+                target.action = Intent.ACTION_PICK
+                target.type = "image/*"
+            }
+        }
+        // Create a chooser from the main  intent
+        val chooserIntent = Intent.createChooser(target, title)
+        // Add all other intents
+        chooserIntent.putExtra(
+            Intent.EXTRA_INITIAL_INTENTS, allIntents.toTypedArray<Parcelable>()
+        )
+        return chooserIntent
+    }
+
+    /**
+     * Get all Camera intents for capturing image using device camera apps.
+     */
+    private fun getCameraIntents(context: Context, packageManager: PackageManager): List<Intent> {
+        val allIntents: MutableList<Intent> = ArrayList()
+        // Determine Uri of camera image to  save.
+        val outputFileUri = getTmpFileUri()
+        latestTmpUri = outputFileUri
+        val captureIntent = Intent(MediaStore.ACTION_IMAGE_CAPTURE)
+        val listCam = packageManager.queryIntentActivities(captureIntent, 0)
+        for (res in listCam) {
```

Comment:
What res? we maybe would benefit from more explicit names

Reply:
Right. I'll fix that